app/models/course_user_datum.rb

```diff
@@ -249,6 +249,88 @@ def ggl_cache_key
     "ggl/dua-#{dua}/u-#{self.id}"
   end
 
+  # This method call is used specifically for the purpose of callback style update to watchlist
+  # Need to archive old instances and also add new instances
+  def update_cud_gdu_watchlist_instances
+    # At this point, all relevant previously cached grace day usage information should be invalidated
+    # Calls to calculate grace day usage should be from scratch
+    
+    # Ignore if this CUD is an instructor or CA
+    return unless self.student?
+    
+    # Get current grace day condition
+    current_conditions = RiskCondition.get_current_for_course(self.course.name)
```

Comment:
i think getting a particular type of condition should be helper functions inside risk condition model